### ISSUE TITLE
CRNN训练后识别率很低

### ISSUE THREAD

**Issue by LMig2k** [chineseocr/chinese-ocr#108]
你好，请教个问题。alphabet =  u'者廖更纳加奉公-'，生成了109312张2个字符的图片, lmdb中train、val数据集相同。CRNN pytorch训练12个小时后结果：
Test loss:0.0067703773384,accuracy:0.99609375,train loss:3.9853101498e-07
然后用train中的**原图片**进行CRNN识别，但识别准确率很低，是不是我哪个环节出问题呢？

训练环境：
1. crnn用的是pytorch实现，pytorch的版本为0.3.1，gpu版本。
2. 注释了代码：crnn_main.py中的val(net, dataset, criterion, max_iter=2)中的
 preds = preds.squeeze(2)   ##165行，参考crnn.pytorch
3. shuffle=True 改为了shuffle=False   ##73行
4. 未加载预训练模型。

修改2，3是因为pytorch的版本为0.3.1。

图片生成方式：text为alphabet中的任一字符同其它一个字符逐一合并得到。30张背景图片，13种比较常规的字体。
def gene(text, fontFile, fontSize, fontColor, bg,  textIndex, fontColorIndex, bgIndex):
    textLen = len(text)
    fileName = str(textIndex)
    fontName = os.path.basename(fontFile)

    imageDir = os.path.join(imageBaseDir, str(textLen), fontName)
    if not os.path.exists(imageDir):
        os.makedirs(imageDir)

    font = pygame.font.Font(fontFile, fontSize)

    widthGap = random.randint(4, maxWidthGap)
    heightGap = random.randint(1, maxHeightGap)
    xBegin = widthGap/2
    yBegin = heightGap/2
    maxHeight = 0

    characterGap = random.randint(3, maxCharacterGap)
    characterRects = []

    characterIndex = 0
    for character in text:
        if characterIndex != 0:
            xBegin += characterGap
        textSurface = font.render(character, True, fontColor)
        textRect = textSurface.get_rect()
        textRect.center = ((textRect.width)/2 + xBegin, (textRect.height)/2 + yBegin)
        xBegin = xBegin + textRect.width
        maxHeight = max(maxHeight, textRect.height)

        characterRects.append((textSurface, textRect))

        characterIndex += 1

    bg_image = pygame.image.load(bg)
    bg_image = pygame.transform.scale(bg_image, (xBegin + widthGap/2, maxHeight + heightGap))

    for characterRect in characterRects:
        bg_image.blit(characterRect[0], characterRect[1])

    filePrefix = fileName + "_" + str(fontSize) + "_" + str(fontColorIndex) + "_" + str(bgIndex)

    pygame.image.save(bg_image, os.path.join(imageDir, filePrefix + imageSuffix))

    file = codecs.open(os.path.join(imageDir, filePrefix + ".txt"), 'w', encoding="utf-8")
    file.write(text)
    file.close()

**Comment by chengchaoyang** [comment]
我的也是用自己生成的数据训练，得到的模型还不如博主提供的原始模型识别率高，不知道是自己训练参数有问题，还是训练数据有问题，希望博主大大可以分享下自己的训练策略，感谢！

**Comment by wenlihaoyu** [comment]
不训练识别10个字符，那么训练的图像长度imgW应该要小一些，如果你取256，那么图像就被过度拉伸，预测的时候，又取的原图像大小，那么预测结果肯定就不理想。当初训练两个字我用的是imgW=48

**Comment by LMig2k** [comment]
@wenlihaoyu 谢谢! 还一个问题，训练2个字符，train，val数据集都是2个字符，识别2个字符图片的准确率比较高，但是识别4个字符图片，只能识别首尾2个字符(准确率也比较高)，中间2字符未识别，你的2个字符训练的模型能识别各种长度字符的图片吗?

**Comment by wenlihaoyu** [comment]
@LMig2k 两个字训练的时候，要注意长度的选择，理论上识别率不会太高。因为像中文词汇或者英文单词，都不止两个字符，训练两个字符，模型捕捉不到这样规律，最好的方式就是训练长字符。这样训练的模型能够识别任意长短字符

**Comment by jiangxiluning** [comment]
@wenlihaoyu 大神，通过换 keys想训练繁体中文，我们生成了100万左右的 10字长度中文字符，就做了模糊和颜色深浅处理，结果模型怎么训练都是 accuary 为0，而且模型 loss非常大 不收敛，有时 val loss 还会上升，之前还是试过在 maxpooling 之前加 dropout2d，发现没什么用。请问这个问题是不是由于 训练集质量处理的过度了？

**Comment by wenlihaoyu** [comment]
检查你训练数据是否有问题。
